services/api/src/modules/payments/payments.service.ts

```diff
@@ -459,23 +462,73 @@ export default class PaymentsService {
     return newPayment
   }
 
-  async getWirePayment(sourceId: string) {
-    // Last 72 hours
-    const newDate72HoursInpast = new Date(
-      new Date().setDate(new Date().getDate() + 3)
+  async handleWirePayment(
+    payment: PaymentModel,
+    trx?: Transaction
+  ): Promise<ToPaymentBase | null> {
+    if (!payment.id || !payment.paymentBankId || !payment.packId) return null
+    // Find bank account in database
+    const foundBankAccount = await PaymentBankAccountModel.query().findById(
+      payment.paymentBankId
     )
+    userInvariant(foundBankAccount, 'bank account was not found', 404)
+
+    // Last 24 hours
+    const newDate24HoursInPast = new Date(
+      new Date().setDate(new Date().getDate() - 1)
+    ).toISOString()
+
     // Get recent payments of wire type
     const payments = await this.circle.getPayments({
-      from: newDate72HoursInpast.toString(),
+      from: newDate24HoursInPast.toString(),
       type: CirclePaymentQueryType.wire,
     })
     if (!payments) return null
+
+    // Retrieve exchange rates for app currency and USD
+    const exchangeRates = await this.coinbase.getExchangeRates({
+      currency: currency.code,
+    })
+    invariant(exchangeRates, 'unable to find exchange rates')
+
     // Find payment with matching source ID
-    const sourcePayments = payments.map(
-      (payment) => payment.source.id === sourceId
-    )
-    userInvariant(sourcePayments, 'payments not found', 404)
-    return sourcePayments
+    const sourcePayment = payments.find((currentPayment) => {
+      // Convert price to USD for payment
+      const amount = convertFromUSD(currentPayment.amount, exchangeRates.rates)
+      invariant(amount !== null, 'unable to convert to currency')
+      const amountInt = formatFloatToInt(amount)
+      return (
+        currentPayment.sourceId === foundBankAccount.externalId &&
+        amountInt === foundBankAccount.amount
+      )
+    })
+    if (!sourcePayment) return null
+    // Update payment details
+    if (payment.status !== sourcePayment.status || !payment.externalId) {
+      await PaymentModel.query(trx).patchAndFetchById(payment.id, {
+        externalId: sourcePayment.externalId,
+        status: sourcePayment.status,
+      })
+    }
+    // Remove claim from pack if payment fails
+    if (sourcePayment.status === PaymentStatus.Failed) {
+      await this.packs.claimPack(
+        {
+          packId: payment.packId,
+          claimedById: null,
+          claimedAt: null,
+        },
+        trx
+      )
+    }
```

Comment:
Should we remove claim if payment fails? Or just mark as a failure and possibly email the client?

Reply:
No strong opinion, but curious what others think. Seems reasonable to email the customer and maybe give them X time to resubmit (maybe 48hrs etc.) before removing the claim info on the pack?

Reply:
Sweet, I like that idea. We could leverage the expiration field already in the database, and have a task that checked for a past expiration date which removed the claim, so that there's only one chance to resubmit. It may need to be longer than 48 since wires take so long. Will have to research and see if I can find how long would be reasonable.

Reply:
Nevermind on the expiration, was thinking of bids. But we could add an expiration field to payments.